### ISSUE TITLE
TestPipelineExceedMaximumServing

### ISSUE THREAD

**Issue by gyuho** [coreos/etcd#6283]
```
--- FAIL: TestPipelineExceedMaximumServing (0.01s)
    pipeline_test.go:100: failed to send out message
```

https://travis-ci.org/coreos/etcd/jobs/155682812#L3382

**Comment by gyuho** [comment]
I've run over 15,000 rounds with no failure. Pretty sure this is just CI being slow, so not able to receive from channel within 10ms. If it happens again, I will reopen this.

**Comment by heyitsanthony** [comment]
```
2016-12-29 19:48:02.842951 I | rafthttp: started HTTP pipelining with peer 1
2016-12-29 19:48:02.843639 I | rafthttp: peer 1 became active
2016-12-29 19:48:02.860835 I | rafthttp: stopped HTTP pipelining with peer 1
2016-12-29 19:48:02.862097 I | rafthttp: started HTTP pipelining with peer 1
2016-12-29 19:48:02.904865 I | rafthttp: stopped HTTP pipelining with peer 1
2016-12-29 19:48:02.906194 I | rafthttp: started HTTP pipelining with peer 1
2016-12-29 19:48:02.930548 I | rafthttp: peer 1 became active
2016-12-29 19:48:02.947504 I | rafthttp: stopped HTTP pipelining with peer 1
--- FAIL: TestPipelineExceedMaximumServing (0.04s)
	pipeline_test.go:82: failed to send out message
	pipeline_test.go:89: unexpected message sendout
```

https://travis-ci.org/coreos/etcd/jobs/187536186#L1759

**Comment by gyuho** [comment]
@xiang90 @heyitsanthony 

Should we just make https://github.com/coreos/etcd/blob/master/rafthttp/pipeline_test.go#L81

```go
for i := 0; i < connPerPipeline+pipelineBufSize; i++ {
	select {
	case p.msgc <- raftpb.Message{}:
	case <-time.After(10 * time.Millisecond):
		t.Errorf("failed to send out message")
	}
}
```

to

```go
for i := 0; i < connPerPipeline+pipelineBufSize; i++ {
	p.msgc <- raftpb.Message{}
}
```

like https://github.com/coreos/etcd/blob/master/rafthttp/pipeline_test.go#L219-L222


As we explained in https://github.com/coreos/etcd/commit/0d35ba9b94369c72e46ed7b57b2bb3427faf66a8

after `pipelineBufSize` gets filled up, it might block in slow CPU systems


I ran over 10K rounds on this test, and can't reproduce...

**Comment by heyitsanthony** [comment]
probably should bump the timeout to 1s like the fix from #6398
